File: go/metric_summary.go
// this file is @generated
package meteroid

import "time"

type MetricSummary struct {
	AggregationKey *string `json:"aggregation_key,omitempty"`

	AggregationType BillingMetricAggregateEnum `json:"aggregation_type"`

	// RFC 3339 timestamp.
	ArchivedAt *time.Time `json:"archived_at,omitempty"`

	Code string `json:"code"`

	// RFC 3339 timestamp.
	CreatedAt time.Time `json:"created_at"`

	Description *string `json:"description,omitempty"`

	Id BillableMetricId `json:"id"`

	Name string `json:"name"`
}

File: python/meteroid/models/metric_summary.py
# this file is @generated
import dataclasses
import typing as t
from datetime import datetime

from ..serialization import BaseModel
from .billable_metric_id import BillableMetricId
from .billing_metric_aggregate_enum import BillingMetricAggregateEnum


@dataclasses.dataclass
class MetricSummary(BaseModel):
    aggregation_type: BillingMetricAggregateEnum

    code: str

    created_at: datetime

    id: BillableMetricId

    name: str

    aggregation_key: t.Optional[str] = None

    archived_at: t.Optional[datetime] = None

    description: t.Optional[str] = None
